### PR TITLE
readme: add chain connection instructions and development section

### PR DESCRIPTION
**Pull request by carlaKC**
This PR updates Faraday's readme to include the latest instructions for running with a bitcoin backend, and adds a development section along with some typo fixes and formatting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,24 +20,36 @@ cd $GOPATH/src/github.com/lightninglabs/faraday
...
 
+### Chain Backend
+Faraday offers node accounting services which require access to a Bitcoin node with `--txindex` set so that it can perform transaction lookup. Currently the `NodeReport` and `CloseReport` endpoints require this connection, and will fail if it is not present. This connection is *optional*, and all other endpoints will function if it is not configured. 
+
+To connect Faraday to bitcoind:
+```
+--connect_bitcoin                       \
+--bitcoin.host={host:port of bitcoind}  \
+--bitcoin.user={bitcoind username}      \
+--bitcoin.password={bitcoind  password}
```
Comment (guggero) [inline README.md:42]: I think you need to set `--httppostmode` for `bitcoind`.

File: README.md [hunk 2]
```diff
@@ -20,24 +20,36 @@ cd $GOPATH/src/github.com/lightninglabs/faraday
...
+```
+
+To connect Faraday to btcd:
+```
+--connect_bitcoin                   \
+--bitcoin.host={host:port of btcd}  \
+--bitcoin.user={btcd username}      \
+--bitcoin.password={btcd password}
```
Comment (guggero) [inline README.md:50]: I think by default `btcd` uses TLS for its RPC connection, so I'd add a `--usetls` and `--tlspath xyz` example flag here.

**Review by guggero — COMMENTED** [review]
Nice addition to the README, I like the contribution section!
To avoid possible issues with the backend connection I think we should make the difference between `bitcoind` and `btcd` a bit more clear.

File: README.md (hunk 1 above)
Comment (carlaKC) [reply README.md:42]: We default to true here, which is cheeky, but btcd just ignores it so I think it's ok.

File: README.md (hunk 2 above)
Comment (carlaKC) [reply README.md]: 🙏 lazy local `--notls` got me here.

**Pull request merged**